tests/test_cookies.py

```diff
@@ -0,0 +1,496 @@
+
+    cj.save(cookie_file)
+    cj.load(cookie_file)
+    assert len(cj.get()) == 6
+
+    # the global cookies are cleared
+    cj.clear()
+    cj.load(cookie_file)
+    assert len(cj.get()) == 5
+
+    cookie = cj.get('.example.com', '/dir_a', 'ccc')
+    assert cookie.name == 'ccc'
+    assert cookie.value == '333'
+    assert cookie.domain == '.example.com'
+    assert cookie.path == '/dir_a/'
+    assert cookie.expires is None
+    assert cookie.secure
+
+
+def test_cookie_jar_iter(cookie_jar):
+    """Test cookie iterator"""
+    count = 0
+    for cookie in cookie_jar.iter():
+        count += 1
+    assert count == 7
+
+    count = 0
+    for cookie in cookie_jar.iter('example.com'):
+        count += 1
+    assert count == 1
+
+    count = 0
+    for cookie in cookie_jar.iter('.example.com', '/dir_a'):
+        count += 1
+    assert count == 1
+
+    with pytest.raises(KeyError):
+        for cookie in cookie_jar.iter('.example2.com'):
+            pass
+
+    with pytest.raises(KeyError):
+        for cookie in cookie_jar.iter('.example.com', '/dir_c'):
+            pass
+
+
+def test_cookie_jar_clear(cookie_jar):
+    """Test clear the cookies in cookie jar"""
+
+    # Clear a non-existing cookie won't raise exception
+    cookie_jar.clear('.example2.com')
+    assert len(cookie_jar.get()) == 7
+
+    # Clear cookies by domain and path
+    cookie_jar.clear('.example.com', '/dir_a')
+    assert len(cookie_jar.get()) == 6
+
+    # Clear global cookies
+    cookie_jar.clear('.')
+    assert len(cookie_jar.get()) == 3
+
+    # Clear all cookies
+    cookie_jar.clear()
+    assert len(cookie_jar.get()) == 0
+
+
+def test_cookie_jar_get_header(cookie_jar):
+    """Test getting the HTTP Cookie header from cookie jar"""
+    header = cookie_jar.get_header('example.com', '/a/b/c', False)
+    assert set(header.split('; ')) == {'g_a=1', 'g_b=2', 'g_c=3', 'aaa=111'}
+
+    header = cookie_jar.get_header('a.example.com', '/dir_a', False)
+    assert set(header.split('; ')) == {'g_a=1', 'g_b=2', 'g_c=3'}
+
+    header = cookie_jar.get_header('a.example.com', '/dir_a/c/d/e', False)
+    assert set(header.split('; ')) == {'g_a=1', 'g_b=2', 'g_c=3'}
+
+    header = cookie_jar.get_header('a.example.com', '/dir_a', True)
+    assert set(header.split('; ')) == {'g_a=1', 'g_b=2', 'g_c=3', 'bbb=222', 'ccc=333'}
+
+    header = cookie_jar.get_header('a.example.com', '/dir_a/c/d/e', True)
+    assert set(header.split('; ')) == {'g_a=1', 'g_b=2', 'g_c=3', 'bbb=222', 'ccc=333'}
+
+    header = cookie_jar.get_header('a.example.com', '/dir_b/dir_c/d/e', True)
+    assert set(header.split('; ')) == {'g_a=1', 'g_b=2', 'g_c=3', 'bbb=222', 'ddd=444'}
+
+    cookie_jar.set(Cookie.create('abc', '123', '.example.com', '/dir_a', max_age=5))
+    header = cookie_jar.get_header('a.example.com', '/dir_a/c/d/e', True)
+    assert set(header.split('; ')) == {'g_a=1', 'g_b=2', 'g_c=3', 'bbb=222', 'ccc=333', 'abc=123'}
+
+    # Wait for the cookie expired
+    time.sleep(5)
```

Comment:
(important)

Even 1s is slowing tests too much.
```suggestion
    time.sleep(1)  # FIXME patch time instead
```